hack/packages/package-tools/cmd/package-bundle-push.go

```diff
@@ -112,3 +119,33 @@ func validatePackageBundlePushFlags() error {
 	}
 	return nil
 }
+
+// prunePackages will update the given repository packages to contain only the
+// bundle packages that match the provided argument(s). If no matching package
+// bundles can be found, an error is returned.
+func prunePackages(repository *Repository, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("at least one package bundle name is required to be specified")
+	}
+
+	var pruned []Package
+	for _, arg := range args {
+		var argFound bool
+		for _, pkg := range repository.Packages {
+			if pkg.Name == arg {
+				argFound = true
+				pruned = append(pruned, pkg)
+			}
+		}
+		if !argFound {
+			fmt.Printf("Warning: unable to find package bundle %q\n", arg)
```

Comment:
The reason I made this a warning and not an error is to accommodate commands like this:

```bash
$ PACKAGE_REPOSITORY="standalone" make push-package-bundles capabilities not-a-valid-bundle standalone-plugins 
```

Notice that there are three bundles being pushed, two that are valid (`capabilities` and `standalone-plugins`), and one is invalid (`not-a-valid-bundle`). With your suggestion, an error will be thrown after `capabilities` is pushed and `standalone-plugins` will not be pushed.

However, the way it is, both `capabilities` and `standalone-plugins` will be pushed and a warning that `not-a-valid-bundle` was not found will be shown as a warning.

I think you understand how users will use our tools better than I do, so I'll leave it up to you if you still want me to stop operations and return an error.

Reply:
pushing happens after the pruning is done, since we are throwing an error during pruning, we would not encounter the situation you just mentioned.. pushing happens in the for loop from [line 70](https://github.com/yharish991/tanzu-framework/pull/2/files#diff-450589cdb9a542f81b2a4057fba635739e7c98861fb658db01244bdd5f33c20aR70)

Reply:
Oh right. My mistake. That means if one bundle does not exist, even if there are valid bundles in the list, none will be pushed. I lean toward pushing what we can and providing a warning of what was skipped.

It seems you prefer only pushing bundles if all of them are valid. In other words, don't push valid bundles if even one in the list is invalid. Am I understanding correctly? If so, I can make the change to return an error.

Reply:
right, i want the users to know that the input they are providing is invalid, else they will never know that one of the package names they supplied is invalid unless they look at the logs in CI system.. and in this case they wouldn't look at the logs because we are not returning an error and they would assume everything went fine and probably never correct their invalid input..

i think in this case it's better to fail-fast rather than fail-safe